src/lib/choice/choice.identifier.ts

```diff
@@ -0,0 +1,27 @@
+export type ChoiceIdentifier<T extends number = number> = T & {
+  readonly brand: unique symbol;
+};
+
+export namespace ChoiceIdentifier {
```

Comment:
Interesting approach! Could you clarify what specific problem this `ChoiceIdentifier` is designed to solve?